src/NServiceBus.Core/Persistence/InMemory/Outbox/InMemoryOutboxStorage.cs

```diff
@@ -31,7 +30,7 @@ public Task Store(OutboxMessage message, OutboxTransaction transaction, ContextB
             var tx = (InMemoryOutboxTransaction) transaction;
             tx.Enlist(() =>
             {
-                if (!storage.TryAdd(message.MessageId, new StoredMessage(message.MessageId, message.TransportOperations.ToList())))
```

Comment:
This copy doesn't seem to do anything, since the original collection and the copy are going to end up pointing to the same objects anyway, so we're just allocating needlessly here.